quinn-proto/src/connection/streams.rs

```diff
@@ -578,18 +586,18 @@ impl Send {
     /// Returns number of unsent bytes and whether the stream was blocked or finishing, indicating
     /// that the application needs to be notified explicitly if the stream was stopped because no
     /// further write calls will be made.
```

Comment:
I decided to change the code here to only propagate the event if necessary, since the required state is available anyway. Made a first pass at an updated comment, let me know if you think this is good enough.

Reply:
It doesn't look like you've pushed any changes yet.

Reply:
Looks good. Passing in the stream's own ID is a little weird, but getting an event iff you need one is nice.